### PR TITLE
Send client-name, client-version query string parameters to completions

### PR DESCRIPTION
**Pull request by dominiccooney**
Closes CODY-1300

Sourcegraph 5.4 or later completions endpoint will start to check a minimum version asserted by the client. This starts sending the relevant header.

Part of sourcegraph/sourcegraph#62116

## Test plan

CI

Note updated recordings have client-name, client-version query string parameters.

### Manual test

Sign in to sg02.sourcegraphcloud.com and do a chat. You should see an error message like "Request Failed: ClientCodyIgnoreCompatibilityError: Cody for vscode version "1.16.0" doesn't match version constraint ">= 1.20.0" Note, the "vscode" and "1.16.0" strings come from the client.

Screenshot 2024-05-02 at 19 26 28

Generate an autocomplete with alt-\ and check the Cody by Sourcegraph output category contains a cody.completion/error log:

```
    "privateMetadata": {
      "message": "Request to https://sg02.sourcegraphcloud.com/.api/completions/code?client-name=vscode&client-version=1.16.0 failed with 406 Not Acceptable: ClientCodyIgnoreCompatibilityError: Cody for vscode version \"1.16.0\" doesn't match version constraint \">= 1.20.0\"\n"
    }
```

JetBrains: Run JetBrains plugin with this agent and do similar actions. Error messages indicate the right client name and version are sent:

Screenshot 2024-05-02 at 19 38 58

### CODE REVIEW

**Review by philipp-spiess — CHANGES_REQUESTED** [review]
Nice! But this misses `lib/shared/src/sourcegraph-api/completions/browserClient.ts` let's make sure to add the headers there as well even though it's unclear wether we actually have customers using it 🤔

File: lib/shared/src/sourcegraph-api/client-name-version.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ */
+export function setClientNameVersion(newClientName: string, newClientVersion: string) {
+    clientName = newClientName
+    clientVersion = newClientVersion
+}
+
+/**
+ * Gets the client info querystring parameters to send with HTTP requests.
+ */
+export function getClientInfoParams(): { [key: string]: string } {
```
[the remark this answers is not in the document]
Comment (dominiccooney) [reply lib/shared/src/sourcegraph-api/client-name-version.ts:23]: I was kinda hoping to discourage people from dumpster diving for specific headers here, but I guess just the type isn't gonna stop them...

**Review by philipp-spiess — APPROVED** [review]
Thank you 🙇

**Review by valerybugakov — APPROVED** [review]
👀

**Pull request merged**